math/check_amicable_pair.cpp

```diff
@@ -0,0 +1,103 @@
+/**
+ * Copyright 2020 @author iamnambiar
+ * 
+ * @file
+ * A C++ Program to check whether a pair of number is amicable pair or not.
+ * 
+ * \brief
+ * Amicable Pair are two different numbers such that sum of the proper divisor 
+ * of each number is equal to the other number.
+ *  
+ * Wikipedia link for Reference: https://en.wikipedia.org/wiki/Amicable_numbers 
+ */
+#include <iostream>
+#include <cassert>
+#include <cmath>
+
+/**
+ * Function to calculate the sum of all the proper divisor 
+ * of an integer.
+ * @param num First number.
+ * @return Sum of the proper divisor of the number.
+ */
+int sum_of_divisor(int num) {
+    // Variable to store the sum of all proper divisors.
+    int sum = 0;
+    // Using sqrt(num) helps to reduce Time complexity by a factor of sqrt(num)
+    for (int div = 2; div <= sqrt(num); ++div) {
+        // Check 'div' is divisor of 'num'.
+        if (num % div == 0) {
+            // If both divisor are same, add once to 'sum'
+            if (div == (num / div)) {
+                sum += div;
+            } else {
+                // If both divisor are not the same, add both to 'sum'.
+                sum += (div + (num / div));
+            }
+        }
+    }
+    return sum + 1;
+}
+
+/**
+ * Function to check whether the pair is amicable or not.
+ * @param x First number.
+ * @param y Second number.
+ * @return true, if the pair is amicable. false, if the pair
+ * is not amicable.
+ */
+bool is_amicable(int x, int y) {
+    if (sum_of_divisor(x) != y) {
+        return false;
+    }
+    return (sum_of_divisor(y) == x);
+}
+
+/**
+ * Function for testing the is_amicable() function with a
+ * first test case of (220, 284) and assert statement.
+ * (220, 284) is an amicable pair.
+ */
+void test1() {
+    // is_amicable(220, 284) returns true.
+    assert(is_amicable(220, 284) == true);
```

Comment:
one line assert functions can be combined in one function with appropriate comments.
Good job on the documetnation

Reply:
In my last code, a reviewer told me to create different functions for each test cases. That is the reason, I did this way. Should I  change this?

Reply:
if the test involves multiple statements, variables, changes to the variables, etc then it would make sense to have each test in a separate function. But if the test is a simple one line check, it would be convenient to have the checks in one `test()` function.